docs/products/access/topics/authentication-methods/authentication-methods.md

```diff
@@ -1,16 +1,188 @@
+ -a,--algorithm <arg>      Algorithm (optional)
+ -i,--id <arg>             Key ID (optional), one will be generated if not
+                           defined
+ -g,--idGenerator <arg>    Key ID generation method (optional). Can be one
+                           of: date, timestamp, sha256, sha1, none. If
+                           omitted, generator method defaults to
+                           'timestamp'.
+ -I,--noGenerateId         <deprecated> Don't generate a Key ID.
+                           (Deprecated, use '-g none' instead.)
+ -p,--showPubKey           Display public key separately (if applicable)
+ -S,--keySet               Wrap the generated key in a KeySet
+ -o,--output <arg>         Write output to file. Will append to existing
+                           KeySet if -S is used. Key material will not be
+                           displayed to console.
+ -P,--pubKeyOutput <arg>   Write public key to separate file. Will append
+                           to existing KeySet if -S is used. Key material
+                           will not be displayed to console. '-o/--output'
+                           must be declared as well.
+ -x,--x509                 Display keys in X509 PEM format
+```
+
+**Example**
+
+```
+java -jar json-web-key-generator.jar -t EC -c P-256 -i 1 -u sig -S -x
+```
+
+The result will be similar to this:
+ 
+```
+{
+    "keys": [
+        {
+            "kty": "EC",
+            "d": "0JUmbEAblKfEfvoYyr1b9RtqmqTW_yExZYsBsgHJMko",
+            "use": "sig",
+            "crv": "P-256",
+            "kid": "1",
+            "x": "ZsitF5jCfGARMx3dip5b62XY0l6_qQm5NZrOKfHu3CQ",
+            "y": "JlIa2T2TVQR2bVFNrjAsxdcsBAi7aPwDp6Dk4cM1CJ4"
+        }
+    ]
+}
+```
+
+Together with this public key:
+
+```
+-----BEGIN PUBLIC KEY-----
+MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEZsitF5jCfGARMx3dip5b62XY0l6/
+qQm5NZrOKfHu3CQmUhrZPZNVBHZtUU2uMCzF1ywECLto/AOnoOThwzUIng==
+-----END PUBLIC KEY-----
+```
+
+The public key should be shared in the Client configuration as a file (`.pem` file) or a `JWKS URI`. Using the `JWKS URI` makes it easier to roll the keys in the future. The `d` value from the example above should be kept private and not end up in the `JWKS URI`.
+
+##### Generating a JWT
+
+Generate a JWT assertion, including the prepared payload, and sign with the generated **private key**. 
+
+#### Making the request to the token endpoint
+
+A client must include the following parameters in a token request when using the `private_key_jwt` method:
+
+| Parameter               | Description               | Type |
+|-------------------------|---------------------------|------|
+| `client_assertion_type` | A type of client_assertion. Its value must be `urn:ietf:params:oauth:client-assertion-type:jwt-bearer`. | String |
+| `client_assertion`      | The signed JWT that we created in the steps above. |
+| `grant_type`.           | Type of the grant used, for example, `client_credentials` |
```

Comment:
for api endpoints we require the token to be in the authorisation header. This is not described. So we see this as a bug? When using a GET method the token will end up in access logs. We would like to avoid that. In that scenario a header approach looks more secure. So even if we are not compliant we need to find a solution for our GET apis.

Reply:
Added an additional section.